assets/src/application/ui/layout/entityList/filterBar/filterState/useFilteredEntities.ts

```diff
@@ -34,13 +34,13 @@ const useFilteredEntities = <D extends string, L extends string, E extends Entit
 			return filteredEntities;
 		}
 		return applySearches(filteredEntities, filterState);
-	}, [cacheIds, searchText]);
+	}, [applySearches, cacheIds, searchText]);
 
 	// sort it
 	cacheIds = entityListCacheIdString(searchResults);
 	const sortedEntities = useMemo<Array<E>>(() => {
 		return applySorters(searchResults, filterState);
-	}, [cacheIds, sortBy]);
+	}, [applySorters, cacheIds, sortBy]);
```

Comment:
Absence of this dependency here was causing that issue, but then adding that dep caused too many re-renders. That's why I wrapped all the callbacks in `useCallback`

Reply:
ya you pretty much have to memoize all callbacks else React generates a new instance every render which triggers further rendering of components downstream